setup.py

```diff
@@ -711,51 +620,11 @@ def srcpath(name=None, suffix=".pyx", subdir="src"):
 # ----------------------------------------------------------------------
 
 
-def setup_package():
-    setuptools_kwargs = {
-        "install_requires": [
-            "python-dateutil >= 2.7.3",
-            "pytz >= 2017.3",
-            f"numpy >= {min_numpy_ver}",
-        ],
-        "setup_requires": [f"numpy >= {min_numpy_ver}"],
```

Comment:
It seems `setup_requires` wasn't added back in setup.cfg? Are there consequences of no longer including it?

Reply:
The added workflow pip installs pandas without the requirements pre-installed. Seems to pull in requirements OK. (I also wanted to test with other versions of setuptools locally though before approving this PR)

OT for the sdist build step we do `python setup.py clean --quiet`. This fails if numpy not installed.

Reply:
But you installed numpy in that workflow, so the question is maybe what would happen if numpy was not installed? (having the setup_requires might give a nicer error message?)

Reply:
That's why I created 2 different conda environments. so numpy is not in the pip-test environment. (In  #36704 the build and test steps are done independently - here I just added a stripped down workflow that we maybe could consider adding permanently to the the ci)